homeassistant/components/keenetic_ndms2/device_tracker.py

```diff
@@ -1,102 +1,175 @@
+    restored = []
+    for entity_entry in registry.entities.values():
+        if (
+            entity_entry.config_entry_id == config_entry.entry_id
+            and entity_entry.domain == DEVICE_TRACKER_DOMAIN
+        ):
+            mac = entity_entry.unique_id.partition("@")[0]
+            if mac not in tracked:
+                tracked.add(mac)
+                restored.append(
+                    KeeneticTracker(
+                        Device(
+                            mac=mac,
+                            # restore the original name as set by the router before
+                            name=entity_entry.original_name,
+                            ip=None,
+                            interface=None,
+                        ),
+                        router,
+                    )
+                )
+
+    if restored:
+        async_add_entities(restored)
+
+    async_dispatcher_connect(hass, router.signal_update, update_from_router)
+
+    return True
+
+
+@callback
+def update_items(router: KeeneticRouter, async_add_entities, tracked: Set[str]):
+    """Update tracked device state from the hub."""
+    new_tracked: List[KeeneticTracker] = []
+    for mac, device in router.last_devices.items():
+        if mac not in tracked:
+            tracked.add(mac)
+            new_tracked.append(KeeneticTracker(device, router))
+
+    if new_tracked:
+        async_add_entities(new_tracked)
+
+
+class KeeneticTracker(ScannerEntity):
+    """Representation of network device."""
+
+    def __init__(self, device: Device, router: KeeneticRouter):
+        """Initialize the tracked device."""
+        self.device = device
+        self.router = router
+        self._last_seen = (
+            dt_util.utcnow() if device.mac in router.last_devices else None
         )
 
-        self.success_init = self._update_info()
-        _LOGGER.info("Scanner initialized")
-
-    def scan_devices(self):
-        """Scan for new devices and return a list with found device IDs."""
-        self._update_info()
+    @property
+    def should_poll(self) -> bool:
+        """Return False since entity pushes its state to HA."""
+        return False
+
+    @property
+    def is_connected(self):
+        """Return true if the device is connected to the network."""
+        return (
+            self._last_seen
+            and (dt_util.utcnow() - self._last_seen)
+            < self.router.consider_home_interval
+        )
 
-        return [device.mac for device in self.last_results]
+    @property
+    def source_type(self):
+        """Return the source type of the client."""
+        return SOURCE_TYPE_ROUTER
+
+    @property
+    def name(self) -> str:
+        """Return the name of the device."""
+        return self.device.name or self.device.mac
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique identifier for this device."""
+        return f"{self.device.mac}@{self.router.config_entry.entry_id}"
```

Comment:
Can there be more than one device with this mac address in the integration?
```suggestion
        return f"{self.device.mac}_{self.router.config_entry.entry_id}"
```

Reply:
Nope. Inside the router, devices are uniquely identified by MAC. Since the integration works with one router the pair of integration unique ID and MAC is sufficiant

Reply:
I was wondering why we add the config entry_id.